### PR TITLE
{lib}[foss/2016b]  MXNet 0.9.3 core library with Python-2.7.12 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by hajgato** [comment]
@boegel Can you tell me what is the problem with the EB name?

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+easyblock = 'MakeCp'
+
+name = 'MXNet'
+version = '0.9.3'
+suffix = '-Python-%(pyver)s'
```
Comment (boegel) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: @hajgato you intended to use `versionsuffix` here?

Comment (hajgato) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: @boegel: khmm. Yes.

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+version = '0.9.3'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://mxnet.io/' 
+description = """Flexible and Efficient Library for Deep Learning"""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+toolchainopts = {'cstd': 'c++11', 'opt': True, 'pic': True, 'openmp': True}
+
+#Download source from github, fix sub branches, checkout to version,
```
Comment (boegel) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: style nitpicking: single space after the `#` (we'll start enforcing that soon with automatic style tests)

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+#Download source from github, fix sub branches, checkout to version, 
+# clean the directory tree, make the tarball
+sources = [SOURCELOWER_TAR_GZ]
+
+patches = ['MXNet-%(version)s_config.patch']
+
+dependencies = [
+    ('Python', '2.7.12'),
+    ('graphviz', '0.5.1', '-Python-%(pyver)s'),
```
Comment (boegel) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: please use `versionsuffix` here

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+name = 'MXNet'
+version = '0.9.3'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://mxnet.io/' 
+description = """Flexible and Efficient Library for Deep Learning"""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+toolchainopts = {'cstd': 'c++11', 'opt': True, 'pic': True, 'openmp': True}
```
Comment (boegel) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: any specific reason you enable `opt` (i.e. `-O3` here)?

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+buildopts += '&& cd python '
+buildopts += '&& env PYTHONPATH=$PYTHONPATH:%(builddir)s/mxnet/lib/python%(pyshortver)s/site-packages/ '
+buildopts += 'python setup.py install --prefix %(builddir)s/mxnet' 
+ 
+files_to_copy = ['lib']
+
+sanity_check_paths = {
+    'files': ['lib/libmxnet.%s' % SHLIB_EXT],
+    'dirs': ['lib/python%(pyshortver)s/site-packages/'],
+}
```
Comment (boegel) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: maybe also add a sanity check command to make sure importing the Python library works?

```python
sanity_check_commands = ["python -c 'import mxnet'"]

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb (hunk 4 above)
Comment (hajgato) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: @ boegel I just followed the original makefile

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb (hunk 2 above)
Comment (hajgato) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: @boegel done

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb (hunk 3 above)
Comment (hajgato) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: @boegel done

File: easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb (hunk 5 above)
Comment (hajgato) [inline easybuild/easyconfigs/m/MXNet/MXNet-0.9.3-foss-2016b-Python-2.7.12.eb]: @boegel done

**Comment by hajgato** [comment]
Better version:
https://github.com/hpcugent/easybuild-easyconfigs/pull/4346

**Pull request closed without merging**